app/src/main/java/com/example/henzoshimada/feeltrip/Mood.java

```diff
@@ -139,51 +158,70 @@ public String getEmotionalState() {
     }
 
     public void setEmotionalState(String emotionalState) {
-        this.emotionalState = emotionalState;
-        setStateByIndex(0);
+        if (!getEmotionalState().equals(emotionalState)) {
+            this.emotionalState = emotionalState;
+            setStateByIndex(0);
+        }
     }
 
     public void setImage(String image) {
-        this.image = image;
-        setStateByIndex(5);
+        if (getImage() == null) {
+            this.image = TextUtils.htmlEncode(image);
+            setStateByIndex(5);
+        } else if (!getImage().equals(image)) {
+            this.image = TextUtils.htmlEncode(image);
+            setStateByIndex(5);
+        }
     }
 
     public String getDescription() {
-        return description;
+        return Html.fromHtml(description).toString();
     }
 
-    public void setDescription(String description, String append) throws DescriptionTooLongException {
-        int count = 0;
-        for (int i = 0; i < description.length(); i++){
-            if (description.charAt(i) == ' '){
-                count++;
+    public void setDescription(String descriptionin) throws DescriptionTooLongException {
+        if (!getDescription().equals(descriptionin)) {
+            int count = 0;
+            String description = descriptionin.trim();
+            for (int i = 0; i < description.length(); i++) {
+                if (description.charAt(i) == ' ') {
+                    count++;
+                }
             }
+            if (description.length() > 20 || count > 2) {
+                throw new DescriptionTooLongException();
+            } else {
+                this.description = TextUtils.htmlEncode(description);
+            }
+            setStateByIndex(1);
         }
-        if (description.length() > 20 || count > 2){
-            throw new DescriptionTooLongException();
-        }else {
-            this.description = description + append;
-        }
-        setStateByIndex(1);
     }
 
     public String getSocialSit() {
         return socialSit;
     }
 
     public void setSocialSit(String socialSit) {
-        this.socialSit = socialSit;
-        setStateByIndex(3);
+        if (!getSocialSit().equals(socialSit)) {
+            this.socialSit = socialSit;
+            setStateByIndex(3);
+        }
     }
 
-    public void setMapPosition(Double latitude, Double longitude){
-        this.latitude = latitude;
-        this.longitude = longitude;
-        this.location = latitude + ", " + longitude;
-        setStateByIndex(6);
+    public void setMapPosition(Double latitude, Double longitude) {
+        if (this.location == null) {
+            this.latitude = latitude;
+            this.longitude = longitude;
+            this.location = latitude + ", " + longitude;
```

Comment:
Only mark location as being changed if the given location is actually different than the stored